drivers/interrupt_controller/intc_shared_irq.c

```diff
@@ -143,54 +143,39 @@ int shared_irq_initialize(const struct device *dev)
 	return 0;
 }
 
-#if CONFIG_SHARED_IRQ_0
-void shared_irq_config_0_irq(void);
-
-const struct shared_irq_config shared_irq_config_0 = {
-	.irq_num = DT_INST_IRQN(0),
-	.client_count = CONFIG_SHARED_IRQ_NUM_CLIENTS,
-	.config = shared_irq_config_0_irq
-};
-
-struct shared_irq_runtime shared_irq_0_runtime;
-
-DEVICE_DT_INST_DEFINE(0, shared_irq_initialize,
-		NULL, &shared_irq_0_runtime,
-		&shared_irq_config_0, POST_KERNEL,
-		CONFIG_SHARED_IRQ_INIT_PRIORITY, &api_funcs);
-
-void shared_irq_config_0_irq(void)
-{
-	IRQ_CONNECT(DT_INST_IRQN(0),
-		    DT_INST_IRQ(0, priority),
-		    shared_irq_isr, DEVICE_DT_INST_GET(0),
-		    DT_INST_IRQ(0, sense));
-}
-
-#endif /* CONFIG_SHARED_IRQ_0 */
-
-#if CONFIG_SHARED_IRQ_1
-void shared_irq_config_1_irq(void);
-
-const struct shared_irq_config shared_irq_config_1 = {
-	.irq_num = DT_INST_IRQN(1),
-	.client_count = CONFIG_SHARED_IRQ_NUM_CLIENTS,
-	.config = shared_irq_config_1_irq
-};
-
-struct shared_irq_runtime shared_irq_1_runtime;
-
-DEVICE_DT_INST_DEFINE(1, shared_irq_initialize,
-		NULL, &shared_irq_1_runtime,
-		&shared_irq_config_1, POST_KERNEL,
-		CONFIG_SHARED_IRQ_INIT_PRIORITY, &api_funcs);
-
-void shared_irq_config_1_irq(void)
-{
-	IRQ_CONNECT(DT_INST_IRQN(1),
-		    DT_INST_IRQ(1, priority),
-		    shared_irq_isr, DEVICE_DT_INST_GET(1),
-		    DT_INST_IRQ(1, sense));
+#define F1(x) 1
+#define INST_SUPPORTS_DEP_ORDS_CNT(index)  \
+	(FOR_EACH(F1, (+), DT_INST_SUPPORTS_DEP_ORDS(index)) - 1)
+
+#define SHARED_IRQ_CONFIG_FUNC(index)					\
+void shared_irq_config_func_##index(void)				\
+{									\
+	IRQ_CONNECT(DT_INST_IRQN(index),				\
+		    DT_INST_IRQ(index, priority),			\
+		    shared_irq_isr,					\
+		    DEVICE_DT_INST_GET(index),				\
+		    COND_CODE_1(DT_INST_IRQ_HAS_CELL(index, sense),	\
+				(DT_INST_IRQ(index, sense)),		\
+				(0)));					\
 }
 
-#endif /* CONFIG_SHARED_IRQ_1 */
+#define SHARED_IRQ_INIT(index)						\
+	SHARED_IRQ_CONFIG_FUNC(index)					\
+	struct shared_irq_client clients[INST_SUPPORTS_DEP_ORDS_CNT(index)]; \
```

Comment:
Doesn't `clients` needs to incorporate `index` into the variable name for multi-instance support?

Reply:
Thanks for this find. Is fixed now.
Additionally using "n" instead of index to avoid overly long line.
Do you think this could still be merge in this release?